### DISCUSSION TITLE
Dehydration Support

### DISCUSSION THREAD

**Discussion by ardissaps** [LordAnaku/Anakus_Status_Bars#4] [Ideas]
Would you please support [Dehydration](https://www.curseforge.com/minecraft/mc-mods/dehydration) mod? Thank you, no hurry

**Comment by LordAnaku** [comment]
Sorry to say that is an issue that would be needed to be brought up with [Dehydration](https://www.curseforge.com/minecraft/mc-mods/dehydration).

I would be unable to add a bar for their mod with out their variables, and would not be able to enable their default bar without breaking my own (at least to my knowledge). Since they have placed their hud render directly in the InGameHud mixin.

**Comment by ardissaps** [comment]
Okay, I posted on their github too.

**Comment by Globox1997** [comment]
Moved the ingamehud mixin to a separate file with [https://github.com/Globox1997/Dehydration/commit/478e829466c14bc4f17c987be1af88c02348552e](https://github.com/Globox1997/Dehydration/commit/478e829466c14bc4f17c987be1af88c02348552e) which can be found here: [https://github.com/Globox1997/Dehydration/blob/1.19/src/main/java/net/dehydration/thirst/ThirstHudRender.java](https://github.com/Globox1997/Dehydration/blob/1.19/src/main/java/net/dehydration/thirst/ThirstHudRender.java)

**Comment by LordAnaku** [comment]
@Globox1997 if you would like to make your mod more compatible with Hud mods, you could use the fabric HudRenderCallback Event as seen [here.](https://github.com/LordAnaku/Anakus_Status_Bars/blob/master/src/main/java/io/github/lordanaku/anakus_status_bars/screen/hud/RenderHudElements.java)

you just need to implement HudRenderCallback and use,
```java
HudRenderCallback.EVENT.register(new name());
```
in onInitializeClient.

**Comment by Globox1997** [comment]
Dehydration exists some time already, not sure when fabric api delivered a callback but I keep it for now as it is :)
Thanks for the suggestion.
